extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeStreamExtractor.java

```diff
@@ -193,74 +193,190 @@ public String getSubChannelAvatarUrl() {
     @Nonnull
     @Override
     public String getDashMpdUrl() {
-        return "";
+        return EMPTY_STRING;
     }
 
     @Nonnull
     @Override
     public String getHlsUrl() {
-        return json.getArray("streamingPlaylists").getObject(0).getString("playlistUrl");
+        assertPageFetched();
+        return json.getObject("files").getString("playlistUrl", EMPTY_STRING);
     }
 
     @Override
-    public List<AudioStream> getAudioStreams() {
-        return Collections.emptyList();
+    public List<AudioStream> getAudioStreams() throws ParsingException {
+        assertPageFetched();
+        final List<AudioStream> audioStreams = new ArrayList<>();
+        // Non-HLS streams
+        try {
+            audioStreams.addAll(getAudioStreamsFromArray(json.getArray("files"), ""));
+        } catch (final Exception ignored) {
+        }
+        // HLS streams
+        try {
+            final JsonArray streamingPlaylists = json.getArray("streamingPlaylists");
+            for (final Object p : streamingPlaylists) {
+                if (!(p instanceof JsonObject)) continue;
+                final JsonObject playlist = (JsonObject) p;
+                final String playlistUrl = playlist.getString("playlistUrl");
+                audioStreams.addAll(getAudioStreamsFromArray(playlist.getArray("files"),
+                        playlistUrl));
+            }
+        } catch (final Exception e) {
+            throw new ParsingException("Could not get video streams", e);
+        }
+
+        return audioStreams;
+    }
+
+    @Nonnull
+    private List<AudioStream> getAudioStreamsFromArray(final JsonArray streams,
+                                                       final String playlistUrl)
+            throws ParsingException {
+        try {
```

Comment:
This is again almost equal to `getVideoStreamsFromArray`, I don't know if it can be improved

Reply:
I tried, but I wasn't able to find a better way with my current knowledge to remove this duplicate code without using unchecked casts.

Reply:
Ok, then keep it this way